nixos/modules/system/boot/unl0kr.nix

```diff
@@ -10,18 +16,58 @@ in
         Whether to enable the unl0kr on-screen keyboard in initrd to unlock LUKS.
       '';
     };
+
+    settings = lib.mkOption {
+      description = ''
+        Configuration for `unl0kr`.
+
+        See `unl0kr.conf(5)` for supported values.
+
+        Alternatively, visit `https://gitlab.com/postmarketOS/buffybox/-/blob/unl0kr-2.0.0/unl0kr.conf`
+      '';
+
+      example = lib.literalExpression ''
+        {
+            general.animations = true;
+            theme = {
+                  default = "pmos-dark";
+                  alternate = "pmos-light";
+            };
+        }
+      '';
+      default = { };
+      type = lib.types.submodule { freeformType = settingsFormat.type; };
+    };
   };
 
   config = lib.mkIf cfg.enable {
-    meta.maintainers = [];
+    meta.maintainers = with lib.maintainers; [ hustlerone ];
     assertions = [
       {
         assertion = cfg.enable -> config.boot.initrd.systemd.enable;
         message = "boot.initrd.unl0kr is only supported with boot.initrd.systemd.";
       }
+      {
+        assertion = !config.boot.plymouth.enable;
+        message = "unl0kr will not work properly if plymouth is enabled.";
+      }
+    ];
+
+    boot.initrd.availableKernelModules = lib.optionals cfg.enable [
+      "hid-multitouch"
+      "hid-generic"
+      "usbhid"
+
+      "i2c-designware-core"
+      "i2c-designware-platform"
+      "i2c-hid-acpi"
+
+      "usbtouchscreen"
```

Comment:
What is the rationale behind these ones specifically? `evdev` seems required, sure. And I suppose `hid-multitouch` and `hid-generic` make pretty obvious sense.

It's not that I'm opposed to adding the others. It just seems like it might be favoring a particular configuration? Feels like we should either be providing a minimal set of modules or a pretty comprehensive set.

Reply:
Though, the change in the size of the initrd is negligible so I think we can just do it like this and forget about it.

Reply:
I'm adding anything that might be related to touch support (The comments in Jovian NixOS suggest that the i2c modules are for touchscreen support on Steam Deck)

Reply:
My intent is to support as much hardware as possible so this module is as painless as possible